app/models/spree/marketing/list/abandoned_cart.rb

```diff
@@ -6,16 +6,14 @@ class AbandonedCart < Spree::Marketing::List
         NAME_TEXT = 'Abandoned Cart'
         AVAILABLE_REPORTS = [:purchases_by].freeze
 
-        def user_ids
```

Comment:
@SergeyMell  You can't just replace `user_ids` method. It will simply raise exception as mentioned [here](https://github.com/vinsol-spree-contrib/spree_marketing/blob/master/app/models/spree/marketing/list.rb#L28) .

If you want to change this list type. You will have need to change the whole flow.

Reply:
The only place where `user_ids` is used is method `users`. As long as I've changed this method it won't break anything. The only thing I should change is 3 test cases - I'll do this ASAP.